stable/redis/templates/_helpers.tpl

```diff
@@ -300,56 +273,79 @@ but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else
+            {{- printf "storageClassName: %s" .Values.persistence.storageClass -}}
         {{- end -}}
     {{- end -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Return  the proper Storage Class for slave
+Return the proper number of Redis instances
 */}}
-{{- define "redis.slave.storageClass" -}}
+{{- define "redis.replicaCount" -}}
+{{- if and .Values.cluster.enabled .Values.cluster.nodes -}}
+replicas: {{ .Values.cluster.nodes }}
+{{- else -}}
+replicas: 1
+{{- end -}}
+{{- end -}}
+
 {{/*
-Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
-but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else logic.
+Renders a value that contains template.
+Usage:
+{{ include "redis.tplValue" ( dict "value" .Values.path.to.the.Value "context" $) }}
 */}}
-{{- if .Values.global -}}
-    {{- if .Values.global.storageClass -}}
-        {{- if (eq "-" .Values.global.storageClass) -}}
-            {{- printf "storageClassName: \"\"" -}}
-        {{- else }}
-            {{- printf "storageClassName: %s" .Values.global.storageClass -}}
-        {{- end -}}
+{{- define "redis.tplValue" -}}
+    {{- if typeIs "string" .value }}
+        {{- tpl .value .context }}
+    {{- else }}
+        {{- tpl (.value | toYaml) .context }}
+    {{- end }}
+{{- end -}}
+
+{{/*
+Determines whether or not to create the Statefulset
+*/}}
+{{- define "redis.createStatefulSet" -}}
+    {{- if not .Values.cluster.enabled -}}
+        {{- true -}}
     {{- else -}}
-        {{- if .Values.slave.persistence.storageClass -}}
-              {{- if (eq "-" .Values.slave.persistence.storageClass) -}}
-                  {{- printf "storageClassName: \"\"" -}}
-              {{- else }}
-                  {{- printf "storageClassName: %s" .Values.slave.persistence.storageClass -}}
-              {{- end -}}
+        {{- if not .Values.cluster.externalAccess.enabled -}}
+            {{- true -}}
         {{- end -}}
-    {{- end -}}
-{{- else -}}
-    {{- if .Values.slave.persistence.storageClass -}}
-        {{- if (eq "-" .Values.slave.persistence.storageClass) -}}
-            {{- printf "storageClassName: \"\"" -}}
-        {{- else }}
-            {{- printf "storageClassName: %s" .Values.slave.persistence.storageClass -}}
+        {{- if and .Values.cluster.externalAccess.enabled .Values.cluster.externalAccess.service.loadBalancerIP -}}
+            {{- true -}}
         {{- end -}}
     {{- end -}}
 {{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "redis.labels" -}}
+app.kubernetes.io/name: {{ include "redis.name" . }}
+helm.sh/chart: {{ include "redis.chart" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end -}}
+
+{{/*
+Labels to use on deploy.spec.selector.matchLabels and svc.spec.selector
+*/}}
+{{- define "redis.matchLabels" -}}
+app.kubernetes.io/name: {{ include "redis.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end -}}
```

Comment:
I definitely see here missing a validateValues helper (we have it in other charts like Cassandra if I'm not mistaken) where we check that the external access configuration is correct. For example, if we have a 6 node cluster we should check that we have 6 load balancer ips.

Reply:
I'm also checking if we can avoid the value externalAccess by just having the service as LoadBalancerIP. We don't have an use case with LoadBalancerIP and not setting externalAccess, do we? I guess that only in the single deployment case but I'm sure we can simplify this.